GenderPayGap.WebUI/Controllers/Admin/AdminDatabaseIntegrityChecksController.cs

```diff
@@ -0,0 +1,201 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using GenderPayGap.Core;
+using GenderPayGap.Core.Classes;
+using GenderPayGap.Core.Interfaces;
+using GenderPayGap.Database;
+using GenderPayGap.WebUI.Models.Admin;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+namespace GenderPayGap.WebUI.Controllers.Admin
+{
+    [Authorize(Roles = "GPGadmin")]
+    [Route("admin")]
+    public class AdminDatabaseIntegrityChecksController : Controller
+    {
+
+        private readonly IDataRepository dataRepository;
+
+        public AdminDatabaseIntegrityChecksController(
+            IDataRepository dataRepository)
+        {
+            this.dataRepository = dataRepository;
+        }
+
+        [HttpGet("database-integrity-checks")]
+        public async Task<IActionResult> DatabaseIntegrityChecks()
+        {
+            return View("DatabaseIntegrityChecks");
+        }
+
+        [HttpGet("organisations-with-multiple-active-addresses-ajax")]
+        public async Task<IActionResult> OrganisationsWithMultipleActiveAddressesAjax()
+        {
+            var model = new DatabaseIntegrityChecksViewModel();
+            List<Organisation> organisations = dataRepository.GetAll<Organisation>()
+                .Include(o => o.OrganisationAddresses)
+                .ToList();
```

Comment:
It might be nice not to do `.ToList()` in all these methods.
That way, the results will be streamed from the database, rather than being read into memory all in one go.

Reply:
Done